hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergedReadHandle.java

```diff
@@ -143,7 +145,7 @@ private List<HoodieRecord<T>> doMergedRead(Option<HoodieFileReader> baseFileRead
     if (baseFileReaderOpt.isPresent()) {
       HoodieFileReader baseFileReader = baseFileReaderOpt.get();
       HoodieRecordMerger recordMerger = config.getRecordMerger();
-      ClosableIterator<HoodieRecord<T>> baseFileItr = baseFileReader.getRecordIterator(readerSchema);
+      ClosableIterator<HoodieRecord<T>> baseFileItr = baseFileReader.getRecordIterator(baseFileReaderSchema);
```

Comment:
@nsivabalan : Wouldn't this cause more columns to be read which is not used for merging ?